src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1048,9 +1050,15 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
 
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
+            ushort lastKeywordIdExclusive = (ushort)SyntaxKind.ElifKeyword;
+
+            for (ushort id = firstKeywordIdInclusive; id < lastKeywordIdExclusive; ++id)
             {
-                yield return (SyntaxKind)i;
+                if (Enum.IsDefined(typeof(SyntaxKind), id))
```

Comment:
Given that, i woudl not make this change.  I would leave the code as is, but add a comment to SyntaxKind on hte last contextual keyword saying "if you add a new contextual keyword, you need to update these helpers: .X, Y Z".